Mathlib/Data/PFunctor/Multivariate/Basic.lean

```diff
@@ -0,0 +1,252 @@
+/-
+Copyright (c) 2018 Jeremy Avigad. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Avigad, Simon Hudon
+
+! This file was ported from Lean 3 source module data.pfunctor.multivariate.basic
+! leanprover-community/mathlib commit e3d9ab8faa9dea8f78155c6c27d62a621f4c152d
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Control.Functor.Multivariate
+import Mathlib.Data.PFunctor.Univariate.Basic
+
+/-!
+# Multivariate polynomial functors.
+
+Multivariate polynomial functors are used for defining M-types and W-types.
+They map a type vector `α` to the type `Σ a : A, B a ⟹ α`, with `A : Type` and
+`B : A → typevec n`. They interact well with Lean's inductive definitions because
```

Comment:
Did you run `scripts/fix_comments.py` on this file? Because `typevec` should be auto-updated to `TypeVec`.

Reply:
```suggestion
`B : A → TypeVec n`. They interact well with Lean's inductive definitions because
```